Source/GMCAbilitySystem/Private/Components/GMCAbilityComponent.cpp

```diff
@@ -449,30 +475,66 @@ void UGMC_AbilitySystemComponent::RPCConfirmAbilityActivation_Implementation(int
 	if (ActiveAbilities.Contains(AbilityID))
 	{
 		ActiveAbilities[AbilityID]->ServerConfirm();
-		UE_LOG(LogGMCAbilitySystem, VeryVerbose, TEXT("[RPC] Server Confirmed Ability Activation: %d"), AbilityID);
+		UE_LOG(LogGMCAbilitySystem, VeryVerbose, TEXT("[RPC] Server Confirmed Long-Running Ability Activation: %d"), AbilityID);
 	}
 }
 
-TSubclassOf<UGMCAbility> UGMC_AbilitySystemComponent::GetGrantedAbilityByTag(FGameplayTag AbilityTag)
+TArray<TSubclassOf<UGMCAbility>> UGMC_AbilitySystemComponent::GetGrantedAbilitiesByTag(FGameplayTag AbilityTag)
 {
 	if (!GrantedAbilityTags.HasTag(AbilityTag))
 	{
 		UE_LOG(LogGMCAbilitySystem, Warning, TEXT("Ability Tag Not Granted: %s"), *AbilityTag.ToString());
-		return nullptr;
+		return {};
 	}
 
 	if (!AbilityMap.Contains(AbilityTag))
 	{
 		UE_LOG(LogGMCAbilitySystem, Warning, TEXT("Ability Tag Not Found: %s | Check The Component's AbilityMap"), *AbilityTag.ToString());
-		return nullptr;
+		return {};
 	}
 
-	return AbilityMap[AbilityTag];
+	return AbilityMap[AbilityTag].Abilities;
 }
 
 void UGMC_AbilitySystemComponent::InitializeAbilityMap(){
-	if(!StartingAbilityMap) return;
-	AbilityMap.Append(StartingAbilityMap.Get()->GetAbilityMap());
+	for (UGMCAbilityMapData* StartingAbilityMap : AbilityMaps)
+	{
+		for (const FAbilityMapData Data : StartingAbilityMap->GetAbilityMapData())
+		{
+			AddAbilityMapData(Data);
+		}
+	}
+}
+
+void UGMC_AbilitySystemComponent::AddAbilityMapData(const FAbilityMapData& AbilityMapData)
+{
+	if (AbilityMap.Contains(AbilityMapData.InputTag))
+	{
+		AbilityMap[AbilityMapData.InputTag] = AbilityMapData;
+	}
+	else
+	{
+		AbilityMap.Add(AbilityMapData.InputTag, AbilityMapData);
+	}
+	
+	if (AbilityMapData.bGrantedByDefault)
+	{
+		GrantedAbilityTags.AddTag(AbilityMapData.InputTag);
+	}
+}
+
+void UGMC_AbilitySystemComponent::RemoveAbilityMapData(const FAbilityMapData& AbilityMapData)
+{
+	if (AbilityMap.Contains(AbilityMapData.InputTag))
+	{
+		AbilityMap.Remove(AbilityMapData.InputTag);
+	}
+	{
+		if (GrantedAbilityTags.HasTag(AbilityMapData.InputTag))
```

Comment:
this is a very small thing but i think this should be .HasTagExact() rather than HasTag. 

![image](https://github.com/reznok/GMCAbilitySystem/assets/33180578/e812686c-5cad-4a2d-a860-0420bdbc9406)

Reply:
For what it's worth, I agree with gilbz on this.

Reply:
I think I agree, though there may be cases where you may want both. Can make the change for now though.

Reply:
Changed for granted/active checks